src/startServer.js

```diff
@@ -0,0 +1,78 @@
+import cluster from 'cluster';
+import os from 'os';
+
+import {
+  CLUSTER_PROCESS_START,
+  CLUSTER_PROCESS_EXIT,
+  SERVER_LISTEN_ERROR,
+} from '#lib/logger.const';
+
+// dotenv should be called on entry to the application to ensure all `process.env.*` variables are correctly set from '.env'
+const dotenv = require('dotenv');
+
+const DOT_ENV_CONFIG = dotenv.config();
+if (DOT_ENV_CONFIG.error) {
+  throw DOT_ENV_CONFIG.error;
+}
+
+// now `process.env.*` variables are set run the rest of the app
+const http = require('http');
+const nodeLogger = require('#lib/logger.node');
+const app = require('./server').default;
+
+const logger = nodeLogger(__filename);
+const port = process.env.PORT || 7080;
+
+const startApplicationInstance = () => {
+  const server = http.createServer(app);
+  let currentApp = app;
+  server.listen(port, error => {
+    if (error) {
+      logger.error(SERVER_LISTEN_ERROR, {
+        error,
+      });
+    }
+  });
+
+  if (module.hot) {
+    logger.info('✅  Server-side Hot Module Replacement enabled');
+
+    module.hot.accept('./server', () => {
+      logger.info('🔁  Hot Module Replacement reloading `./server`...');
+      server.removeListener('request', currentApp);
+      const newApp = require('./server').default; // eslint-disable-line global-require
+      server.on('request', newApp);
+      currentApp = newApp;
+    });
+  }
+};
+
+const processOnline = worker =>
+  logger.info(CLUSTER_PROCESS_START, `Worker ${worker.id} started`);
+
+const processExit = (worker, code, signal) => {
+  const exitReason = code ? ` with code ${code}` : ` due to signal ${signal}`;
+
+  if (code !== 0 && !worker.exitedAfterDisconnect) {
+    cluster.fork();
+  }
+
+  return logger.error(
+    CLUSTER_PROCESS_EXIT,
+    `Worker ${worker.id} terminated ${exitReason}`,
+  );
+};
+
+const startCluster = () => {
+  if (!cluster.isWorker) {
+    const availableCores = os.cpus();
+    availableCores.map(() => cluster.fork());
+    cluster.on('online', processOnline);
+    cluster.on('exit', processExit);
+  } else {
+    startApplicationInstance();
+  }
+};
```

Comment:
Am I right in saying, we are enabling cluster workers in local development too? Will that cause any issues?

Reply:
It will, it shouldn't cause any issues locally though it might work your machine a tiny bit harder...

Reply:
we could always implement an env variable override option if we wanted it disabled on localhost

Reply:
added in latest commit